[gtest] debug build failure

### Operating system

Windows

### Compiler

MSVC

### Steps to reproduce the behavior

Just build only `VCPKG_BUILD_TYPE=debug` and observe how it fails.

The problem is a typo in the `portfile.cmake`:
https://github.com/microsoft/vcpkg/blob/61c70ce627a2a4ffff809f887e4188ff39ce0a14/ports/gtest/portfile.cmake#L56-L63

It attempts to patch the release build pkg-config files in the debug case. It should be `VCPKG_BUILD_TYPE STREQUAL "release"` in the first case.


### Failure logs

```
-- Fixing pkgconfig file: C:/Users/dsven/CLionProjects/foo/vcpkg/packages/gtest_x64-windows-static-asan/debug/lib/pkgconfig/gmock.pc
-- Fixing pkgconfig file: C:/Users/dsven/CLionProjects/foo/vcpkg/packages/gtest_x64-windows-static-asan/debug/lib/pkgconfig/gmock_main.pc
-- Fixing pkgconfig file: C:/Users/dsven/CLionProjects/foo/vcpkg/packages/gtest_x64-windows-static-asan/debug/lib/pkgconfig/gtest.pc
-- Fixing pkgconfig file: C:/Users/dsven/CLionProjects/foo/vcpkg/packages/gtest_x64-windows-static-asan/debug/lib/pkgconfig/gtest_main.pc
CMake Error at scripts/cmake/vcpkg_replace_string.cmake:12 (file):
  file failed to open for reading (No such file or directory):

    C:/Users/dsven/CLionProjects/foo/vcpkg/packages/gtest_x64-windows-static-asan/lib/pkgconfig/gmock_main.pc
```

### Additional context

_No response_

--- a/ports/gtest/portfile.cmake
+++ b/ports/gtest/portfile.cmake
@@ -53,7 +53,7 @@ file(
 file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/include)
 
 vcpkg_fixup_pkgconfig()
-if(NOT DEFINED VCPKG_BUILD_TYPE OR VCPKG_BUILD_TYPE STREQUAL "debug")
+if(NOT DEFINED VCPKG_BUILD_TYPE OR VCPKG_BUILD_TYPE STREQUAL "release")
     vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/lib/pkgconfig/gmock_main.pc" "libdir=\${prefix}/lib" "libdir=\${prefix}/lib/manual-link")
     vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/lib/pkgconfig/gtest_main.pc" "libdir=\${prefix}/lib" "libdir=\${prefix}/lib/manual-link")
 endif()

--- a/ports/gtest/vcpkg.json
+++ b/ports/gtest/vcpkg.json
@@ -1,6 +1,7 @@
 {
   "name": "gtest",
   "version-semver": "1.17.0",
+  "port-version": 1,
   "description": "Google Testing and Mocking Framework",
   "homepage": "https://github.com/google/googletest",
   "license": "BSD-3-Clause",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -3382,7 +3382,7 @@
     },
     "gtest": {
       "baseline": "1.17.0",
-      "port-version": 0
+      "port-version": 1
     },
     "gtk": {
       "baseline": "4.16.3",

--- a/versions/g-/gtest.json
+++ b/versions/g-/gtest.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "9ef57e3c1c7484e79ef8dd77e9cb7770f5331248",
+      "version-semver": "1.17.0",
+      "port-version": 1
+    },
+    {
       "git-tree": "8379b242dedcaefa81bebb7dc499cd6b62a740b4",
       "version-semver": "1.17.0",
       "port-version": 0

